pkg/utils/hex/hex.go

```diff
@@ -0,0 +1,55 @@
+package hex
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/big"
+	"strings"
+)
+
+// EnsurePrefix adds the prefix (0x) to a given hex string.
+func EnsurePrefix(str string) string {
+	if !strings.HasPrefix(str, "0x") {
+		str = "0x" + str
+	}
+	return str
+}
+
+// ToBig parses the given hex string and returns error if it is invalid.
+func ToBig(s string) (*big.Int, error) {
```

Comment:
WDYT about call this `ParseBig`?
```suggestion
// ParseBig parses the given hex string and returns error if it is invalid.
func ParseBig(s string) (*big.Int, error) {
```

Reply:
TBH `ParseBig` sounds like a function that parses big numbers to hex. Maybe `ParseToBig` instead?

Reply:
I really think that `hex.ParseBig` is the clear idiomatic name. The signature and the package already tell the rest of the story. Here are some std lib examples:
- https://pkg.go.dev/net/url#ParseQuery
- https://pkg.go.dev/net/url#Parse
- https://pkg.go.dev/github.com/google/uuid#Parse
- https://pkg.go.dev/time#Parse
- https://pkg.go.dev/time#ParseDuration
- https://pkg.go.dev/net/http#ParseTime